Parse/src/main/java/com/parse/ParseLogInterceptor.java

```diff
@@ -93,67 +109,60 @@ public void writeLine(String str) {
   }
 
   private static class ProxyInputStream extends InputStream {
+    // Helper stream to proxy the original input stream to other input stream
+    private final InputStream originalInput;
+    private PipedInputStream proxyInput;
+    private PipedOutputStream proxyOutput;
```

Comment:
You can get around the problem with final and try/catch by using a local var in the constructor and setting the instance vars after the try/catch.